tests/deploy/e2e/test_deploy.py

```diff
@@ -1068,3 +1068,78 @@ async def test_skipped_for_dependency(resource_container, server, client, client
         deployment_result=DeploymentResult.DEPLOYED,
         blocked=BlockedStatus.NO,
     )
+
+
+async def test_redeploy_after_dependency_recovered(resource_container, server, client, clienthelper, environment, agent):
+    """
+    Asserts that a transiently skipped resource recovers when its dependencies succeed
+    """
+    version = await clienthelper.get_version()
+
+    # Disable auto deploys and repairs
+    Config.set("config", "agent-repair-interval", "0")
+    Config.set("config", "agent-deploy-interval", "0")
+
+    resource_container.Provider.set_fail("agent1", "key", 1)
+
+    rid1 = "test::Resource[agent1,key=key]"
+    rid2 = "test::Resource[agent1,key=key2]"
+
+    resources = [
+        {
+            "key": "key",
+            "value": "value",
+            "id": f"{rid1},v={version}",
+            "requires": [],
+            "purged": False,
+            "send_event": True,
+            "receive_events": False,
+        },
+        {
+            "key": "key2",
+            "value": "value",
+            "id": f"{rid2},v={version}",
+            "requires": [rid1],
+            "purged": False,
+            "send_event": True,
+            "receive_events": True,
+        },
+    ]
+    await clienthelper.set_auto_deploy(True)
+    await clienthelper.put_version_simple(resources, version, wait_for_released=True)
+
+    await clienthelper.wait_for_deployed()
+    scheduler = agent.scheduler
+    assert scheduler._state.resource_state[rid2] == ResourceState(
+        status=ComplianceStatus.NON_COMPLIANT,
+        deployment_result=DeploymentResult.SKIPPED,
+        blocked=BlockedStatus.TRANSIENT,
+    )
+    assert scheduler._state.resource_state[rid1] == ResourceState(
+        status=ComplianceStatus.NON_COMPLIANT,
+        deployment_result=DeploymentResult.FAILED,
+        blocked=BlockedStatus.NO,
+    )
```

Comment:
I'm not following with how you make this fail. And how you make it succeed later on.

Reply:
This is still unclear to me. What am I missing?